### PR TITLE
FMS_interface override FMS constants with CESM share constants

### PR DESCRIPTION
**Pull request by jtruesdal**
This PR allows the FMS to use CESM share global constants in place of the original values.  The change allows MOM and FV3 to run with values that are consistent across the CESM framework.  A new constants file is compiled in place of the original FMS file.  A user may choose to use the original FMS constants by setting the CPP variable USE_FMSCONST. If this is not set, FMS will use the CESM share constants by default.  I modified Makefile.cesm to include the share .mod files when compiling FMS.  This could also be accomplished via buildlib.  I tested the new CESM share constants using the FV3 dycore and Alper tested the constants with MOM.

### CODE REVIEW

File: src_override/constants.F90 [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+! Include variable "version" to be written to log file.
+#include<file_version.h>
+!-----------------------------------------------------------------------
+! version is public so that write_version_number can be called for constants_mod
+! by fms_init
+public :: version
+
+real :: realnumber !< dummy variable to use in HUGE initializations
+
+!-----------------------------------------------------------------------
```
Comment (jedwards4b) [inline src_override/constants.F90:176]: You are creating new paramater variables here when all you need to do is rename the existing ones.
Please changes this, for example:
use shr_const_mod, only RADIUS=>SHR_CONST_REARTH

**Comment by jtruesdal** [comment]
Sorry, I'm still having some problems with this.  Can you unmerge easily or
I can try to get it fixed after my 9:00 meeting.
jt

**Comment by jedwards4b** [comment]
It appears that you have variable declarations preceeding use statements.   I don't think that this repo gets enough traffic that we need to unmerge, please fix as soon as you can.

**Comment by jtruesdal** [comment]
will do.

**Comment by jtruesdal** [comment]
Maybe I'm not implementing this correctly.  The routines that use constants
from this file are not seeing those that are declared as a pointer.  It
looks like I need all the parameter statements.

jt

**Comment by jtruesdal** [comment]
Should I just push my original constants file back?

**Comment by jedwards4b** [comment]
No, you just need to make them public, so add
public :: PI, PI_8, ...

**Comment by jtruesdal** [comment]
Sorry, I need to get more sleep.  Ok, let's try again.  This time I made sure all my tests finished before pushing to the repo. 

Do you know when a new FMS_interface tag will be made?  I need to add that to my FV3 PR.

**Comment by jedwards4b** [comment]
Since I already merged you will need to create a new PR - I can make a tag as soon as it is merged.

**Comment by jtruesdal** [comment]
OK

**Pull request merged**